src/deposit.cairo

```diff
@@ -85,37 +88,71 @@ mod Deposit {
         PositionClosed: PositionClosed
     }
 
-    #[abi(embed_v0)]
-    impl Deposit of IDeposit<ContractState> {
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
         fn swap(ref self: ContractState, swap_data: SwapData) -> SwapResult {
             ekubo::components::shared_locker::call_core_with_callback(
                 self.ekubo_core.read(), @swap_data
             )
         }
+    }
 
+    #[abi(embed_v0)]
+    // TODO: Change types to aliases
+    impl Deposit of IDeposit<ContractState> {
+        /// Loops collateral token on ZKlend.
+        ///
+        /// # Panics
+        ///
+        /// `is_position_open` storage variable is set to true('Open position already exists')
+        /// `amount` field of `deposit_data` is `0` or `pool_price` is `0`
+        /// address of the caller is not equal to `owner` storage address
+        ///
+        /// # Paraemters
+        /// * `deposit_data` - Object of which stores main deposit information.
+        /// * `pool_key` - Ekubo type which represents data about pool.
+        /// * `pool_price` - Price of `deposit` token in terms of `debt` token in Ekubo pool.
+        /// * `usdc_price` - Price of `deposit` token in USDC.
         fn loop_liquidity(
             ref self: ContractState,
             deposit_data: DepositData,
             pool_key: PoolKey,
-            pool_price: felt252
+            pool_price: u256,
+            usdc_price: u256
         ) {
-            assert(get_caller_address() == self.owner.read(), 'Caller is not an owner');
+            // TODO: Add borrow factor
+            let user_acount = get_tx_info().unbox().account_contract_address;
```

Comment:
Why not get_caller_address() ? (Maybe it was removed in a new Cairo version that I'm not yet aware of?)

Reply:
It was made because of testing troubles. In a contract, when swapping we are getting [callback](https://github.com/djeck1432/spotnet/blob/main/src/deposit.cairo#L91) from Ekubo. When using normal cheat_caller_address, it cheats all input calls to the contract, and code fails with 'Caller is not the core' error from Ekubo core contract, so I cheat the account that initiated transaction instead(practically it should be the caller themself). However I agree this functionality is an unusual and unpredicted, so I should include it in the docs.

Reply:
I think okay to just add an explainer comment like this to the code. Interesting!